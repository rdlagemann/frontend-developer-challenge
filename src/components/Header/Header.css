.Header {
  color: white;
  text-align: center;
  width: 100%;
  display: flex;
  justify-content: flex-end;
  flex-direction: column;
  align-items: center;
  padding: 30px;
  position: relative;
}

.Header-headline {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.Header__text {
  font-size: 14px;
  margin-bottom: 30px;
}

.Header__text-big {
  font-size: 40px;
  line-height: 49px;
  margin-bottom: 24px;
}

.Header__text-small {
  font-size: 20px;
  font-weight: normal;
  line-height: 24px;
}
