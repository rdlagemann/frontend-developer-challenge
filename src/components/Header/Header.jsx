import React from 'react'
import PropTypes from 'prop-types'
import './Header.css'
import Nav from './Nav'
import cx from 'classnames'
import Menu from '../Menu'

Header.propTypes = {
  className: PropTypes.string,
  headline: PropTypes.array,
  navLinks: PropTypes.array
}

Header.defaultProps = {
  className: '',
  headline: [],
  navLinks: []
}

export default function Header({ className, headline, description, navLinks }) {
  return (
    <header className={cx('Header', className)}>
      <div className="container">
        {!!headline.length && (
          <h1 className="Header-headline" data-test="headline">
            <span className="Header__text-small">{headline[0]}</span>{' '}
            <span className="Header__text-big">{headline[1]}</span>
          </h1>
        )}
        {!!description && (
          <p className="Header__text" data-test="description">
            {description}
          </p>
        )}

        <Menu className="d-flex d-md-none" links={navLinks} />
        <Nav className="d-none d-md-block" links={navLinks} />
      </div>
    </header>
  )
}
