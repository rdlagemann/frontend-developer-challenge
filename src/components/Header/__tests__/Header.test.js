import React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import Header from '..'

describe('Header', () => {
  it('renders', () => {
    const output = create(<Header />)
    expect(output).toMatchSnapshot()
  })

  it('should contain a single headline', () => {
    const headline = ['produtos especiais para você']
    const headerWrapper = shallow(<Header headline={headline} />)
    expect(
      headerWrapper
        .find('[data-test="headline"]')
        .text()
        .trim()
    ).toBe(headline.join(' '))
  })

  it('should contain a double headline', () => {
    const headline = ['uma seleção de produtos', 'especial para você']
    const headerWrapper = shallow(<Header headline={headline} />)
    expect(
      headerWrapper
        .find('[data-test="headline"]')
        .text()
        .trim()
    ).toBe(headline.join(' '))
  })

  it('should NOT contain a headline', () => {
    const headline = []
    const headerWrapper = shallow(<Header headline={headline} />)
    expect(headerWrapper.find('[data-test="headline"]')).toEqual({})
  })

  it('should contain a description', () => {
    const description = 'Descrição teste'
    const headerWrapper = shallow(<Header description={description} />)

    expect(
      headerWrapper
        .find('[data-test="description"]')
        .text()
        .trim()
    ).toBe(description)
  })
})
