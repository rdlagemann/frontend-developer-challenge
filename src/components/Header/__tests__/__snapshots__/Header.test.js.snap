// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders 1`] = `
<header
  className="Header"
>
  <div
    className="container"
  >
    <button
      className="Hamburguer d-block d-md-none"
      onClick={[Function]}
    >
      <img
        alt="Menu"
        className="Hamburguer__icon"
        src="menu.svg"
      />
    </button>
    <div
      className="Menu d-flex d-md-none"
    >
      <span
        className="Menu__list"
      />
    </div>
  </div>
</header>
`;
